### ISSUE TITLE
Supporting AsyncDisplayKit (Texture)

### ISSUE THREAD

**Issue by JoeFerrucci** [johnno1962/InjectionIII#435]
[AsyncDisplayKit](https://texturegroup.org/) (ASDK), also known as [Texture](https://texturegroup.org/).

InjectionIII doesn't support making changes whenever ASDK is involved, particularly in subclasses of ASDK classes like ASCellNode or ASDisplayNode, etc. Even if my subclass is marked `final`. It also doesn't matter the location of the change or how simple or complex it is: changing a String, add a print statement in a function, changing the value of a property of another object. 

Literally ANY change made in or around ASDK causes Injection to crash the app. 

Once I leave the screen in which the class is being used and take action in my app to use the edited code, I get the error and crash

Here is an example of the failure I get from the console as well as from Xcode:

## Console:
```bash
libc++abi: libc++abi: libc++abi: libc++abi: terminating with uncaught exception of type NSExceptionlibc++abi: terminating with uncaught exception of type NSException
terminating with uncaught exception of type NSException
libc++abi: terminating with uncaught exception of type NSException
libc++abi: terminating with uncaught exception of type NSException
terminating with uncaught exception of type NSException
libc++abi: terminating with uncaught exception of type NSException
libc++abi: terminating with uncaught exception of type NSException

terminating with uncaught exception of type NSException
warning: hit breakpoint while running function, skipping commands and conditions to prevent recursion
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Node class MyApp.MyCustomCellNode does not have a matching _staticInitialize method; check to ensure [super initialize] is called within any custom +initialize implementations!  Overridden methods will not be called unless they are also implemented by superclass MyApp.MyCustomCellNode'
terminating with uncaught exception of type NSException
CoreSimulator 857.14 - Device: iPhone 14 Pro (MyUUID) - Runtime: iOS 16.2 (20C52) - DeviceType: iPhone 14 Pro
```

## Xcode:
```bash
libsystem_kernel.dylib:
Thread 54: "Node class MyApp.MyCustomCellNode does not have a matching _staticInitialize method; check to ensure [super initialize] is called within any custom +initialize implementations!  Overridden methods will not be called unless they are also implemented by superclass MyApp.MyCustomCellNode"
```

**Comment by johnno1962** [comment]
Hi, I've never had this reported before and apart from the error message I wouldn't know where to start. Is there any chance you would be able to put together a simple example app using ASDK that demonstrates the problem and post it here or email it to me at github at johnholdsworth.com. Once I have been able to replicate it on my machine I will be able to take a look for you.

**Comment by johnno1962** [comment]
Looking at ASDK, can you try commenting out this asset and see how far you get:
```
    ASDisplayNodeAssert(node.class == initializeSelf, @"Node class %@ does not have a matching _staticInitialize method; check to ensure [super initialize] is called within any custom +initialize implementations!  Overridden methods will not be called unless they are also implemented by superclass %@", node.class, initializeSelf);
```
I doubt I'll be able to debug it this end as I don't have CocoaPods installed on my Mac.

**Comment by JoeFerrucci** [comment]
Emailed you a basic project. ~1.3MB with the Texture pod.

**Comment by JoeFerrucci** [comment]
It works when I comment out that assertion line, but that requires me to unlock the texture pod and manually edit it.

**Comment by johnno1962** [comment]
Thanks for the example project. I was able to run it but I can't replicate the problem you're seeing. Which version of InjectionIII are you using? Have you tried the most recent github binary release in this repo?

**Comment by JoeFerrucci** [comment]
> Thanks for the example project. I was able to run it but I can't replicate the problem you're seeing. Which version of InjectionIII are you using? Have you tried the most recent github binary release in this repo?

I believe so. Which version do you see as latest?

**Comment by johnno1962** [comment]
Hi, just released 4.5.7 today though that shouldn't bee the problem you're seeing. I tried your example project (thanks!) but couldn't get it to crash in the way you're saying by uncommenting the line you mention.

**Comment by johnno1962** [comment]
Thanks, I'll leave it open for now if you're still interested in a potential fix eventually though ASDK may be a bad mix with injection - They are up to a lot of tricky swizzling stuff.
